retryrpc/server.go

```diff
@@ -452,32 +484,31 @@ func (server *Server) trimAClientBasedACK(uniqueID string, ci *clientInfo) (numI
 // This gets called every ~10 minutes to clean out older entries.
 //
 // NOTE: We assume Server Lock is held
-func (server *Server) trimTLLBased(uniqueID string, ci *clientInfo, t time.Time) (numItems int) {
-
-	l := list.New()
-
+func (server *Server) trimTLLBased(ci *clientInfo, t time.Time) (numItems int) {
 	ci.Lock()
-	for e := ci.completedRequestLRU.Front(); e != nil; e = e.Next() {
+	for e := ci.completedRequestLRU.Front(); e != nil; {
 		eTime := e.Value.(completedLRUEntry).timeCompleted.Add(server.completedLongTTL)
 		if eTime.Before(t) {
 			delete(ci.completedRequest, e.Value.(completedLRUEntry).requestID)
+			ci.stats.cntRmCompleted++
 
-			// Push on local list so don't delete while iterating
-			l.PushBack(e)
+			eTmp := e
+			e = e.Next()
+			_ = ci.completedRequestLRU.Remove(eTmp)
+			numItems++
 		} else {
 			// Oldest is in front so just break
 			break
 		}
 	}
+	s := ci.stats
+	logger.Infof("ID: %v completedRequest len: %v completedRequestLRU len: %v previousHighestReplySeen: %v highestReplySeen: %v "+
```

Comment:
I see what you did here... You have your own "stats" struct that you rapidly update as operations occur... then, here, you "log" the results.

Can I say that I absolutely HATE logging stats? That is something a tool outside ProxyFS et. al. should do by scraping /stats and/or /metrics imho. Just my $0.02.

But what happened to using bucketstats (let alone stats)?

Reply:
I am removing this logging it my next commit.   I wanted to get this merged first